Description: revert commit 735bd2a53834266c7256830c8d34672ea55fe17b
Author: Marc Deslauriers <marc.deslauriers@canonical.com>
Forwarded: No, not needed
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/mysql-8.0/+bug/1889851

--- a/cmake/build_configurations/compiler_options.cmake
+++ b/cmake/build_configurations/compiler_options.cmake
@@ -71,12 +71,6 @@ IF(UNIX)
     SET(COMMON_CXX_FLAGS             "-std=c++14 -fno-omit-frame-pointer")
   ENDIF()
 
-  # Faster TLS model
-  IF(MY_COMPILER_IS_GNU_OR_CLANG AND NOT SOLARIS AND NOT LINUX_RHEL6)
-    STRING_APPEND(COMMON_C_FLAGS     " -ftls-model=initial-exec")
-    STRING_APPEND(COMMON_CXX_FLAGS   " -ftls-model=initial-exec")
-  ENDIF()
-
   # Solaris flags
   IF(SOLARIS)
     # Link mysqld with mtmalloc on Solaris 10 and later
